### ISSUE TITLE
Style/RedundantRegexpEscape flags escaped delimeters in %r regexps

### ISSUE THREAD

**Issue by pyzlnar** [rubocop-hq/rubocop#8131]
## Expected behavior

The [`Style/RedundantRegexpEscape`](https://docs.rubocop.org/rubocop/0.85/cops_style.html#styleredundantregexpescape) cop should not flag or fix escaped characters when they are being used as delimiters in a `%r` type regexp.

## Actual behavior

The `Style/RedundantRegexpEscape` cop flags and fixes escaped characters when they are being used as delimiters in a `%r` type regexp. This causes a Syntax Error under several ruby versions. 

## Steps to reproduce the problem

Take this regexp matching for `/` or `}` characters:

```ruby
%r{[/\}]}
```

Currently the cop will register this as an offense, and if run with autocorrect creates a syntax error.

```bash
$ rubocop test_file.rb -a --only Style/RedundantRegexpEscape
Inspecting 1 file
E

Offenses:

test_file.rb:1:1: E: Lint/Syntax: premature end of char-class: /[\//
(Using Ruby 2.6 parser; configure using TargetRubyVersion parameter, under AllCops)
%r{[/}]}
^^^^^^
test_file.rb:1:6: C: [Corrected] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r{[/\}]}
     ^^

1 file inspected, 2 offenses detected, 1 offense corrected
```

Keep in mind that this applies for any delimiter, whether it be braces `{ }` or other less common delimiters.

```ruby
# Other contrived examples

# Opening Brace
%r{[/\{]}

# Closing Parenthesis
%r([/\)])

# Pipe
%r|[/\|]|

# Outside brackets
%r-a\-b-

# etc...
%r~[/\~]~
```

```
rubocop test_file.rb --only Style/RedundantRegexpEscape
Inspecting 1 file
C

Offenses:

test_file.rb:2:6: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r{[/\{]}
     ^^
test_file.rb:5:6: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r([/\)])
     ^^
test_file.rb:8:6: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r|[/\|]|
     ^^
test_file.rb:11:5: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r-a\-b-
    ^^
test_file.rb:14:6: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
%r~[/\~]~
     ^^

1 file inspected, 5 offenses detected
```

## RuboCop version

```bash
$ rubocop -V
0.85.1 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.6.3 x86_64-linux)
```
